mvcc/index.go

```diff
@@ -208,6 +209,34 @@ func (ti *treeIndex) Compact(rev int64) map[revision]struct{} {
 	return available
 }
 
+// Compact2 is different from Compact, the return map is the data going to be deleted in backend
```

Comment:
Need to declare what both return maps are, e.g.:

```
// Compact2 compacts the index and returns the revisions and tombstone revisions that were compacted.
```

Out of curiosity, does the return types really need to be sets, or can they be slices now?

Reply:
slice is ok